src/modules/powerrename/lib/Settings.cpp

```diff
@@ -52,34 +57,208 @@ namespace
         SetRegNumber(valueName, value ? 1 : 0);
     }
 
-    std::wstring GetRegString(const std::wstring& valueName) {
+    std::wstring GetRegString(const std::wstring& valueName,const std::wstring& subPath)
+    {
         wchar_t value[CSettings::MAX_INPUT_STRING_LEN];
         value[0] = L'\0';
         DWORD type = REG_SZ;
         DWORD size = CSettings::MAX_INPUT_STRING_LEN * sizeof(wchar_t);
-        if (SUCCEEDED(HRESULT_FROM_WIN32(SHGetValue(HKEY_CURRENT_USER, c_rootRegPath, valueName.c_str(), &type, value, &size) == ERROR_SUCCESS)))
-        {
-            return std::wstring(value);
-        }
-        return std::wstring{};
+        std::wstring completePath = std::wstring(c_rootRegPath) + subPath;
+        SHGetValue(HKEY_CURRENT_USER, completePath.c_str(), valueName.c_str(), &type, value, &size);
+        return std::wstring(value);
+    }
+
+    FILETIME LastModifiedTime(const std::wstring& filePath)
+    {
+        WIN32_FILE_ATTRIBUTE_DATA attr{};
+        GetFileAttributesExW(filePath.c_str(), GetFileExInfoStandard, &attr);
+        return attr.ftLastWriteTime;
+    }
+}
+
+class MRUListHandler
+{
+public:
+    MRUListHandler(int size, const std::wstring& filePath, const std::wstring& regPath) :
+        pushIdx(0),
+        nextIdx(1),
+        size(size)
+    {
+        items.resize(size);
+        std::wstring result = PTSettingsHelper::get_module_save_folder_location(L"PowerRename");
+        jsonFilePath = result + L"\\" + filePath;
+        registryFilePath = regPath;
+        Load();
+    }
+
+    void Push(const std::wstring& data);
+    bool Next(std::wstring& data);
+
+    void Reset();
+
+private:
+    void Load();
+    void Save();
+    void MigrateFromRegistry();
+    json::JsonArray Serialize();
+    void ParseJson();
+
+    bool Exists(const std::wstring& data);
+
+    std::vector<std::wstring> items;
+    long pushIdx;
+    long nextIdx;
+    long size;
+    std::wstring jsonFilePath;

```

Comment:
`jsonFilePath` could be const and initialized in member initializer list of the constructor

Reply:
Done.